gson/src/main/java/com/google/gson/internal/bind/CollectionTypeAdapterFactory.java

```diff
@@ -58,6 +64,20 @@ public CollectionTypeAdapterFactory(ConstructorConstructor constructorConstructo
     return result;
   }
 
+  private boolean isRawTypeSupported(Class<?> rawType) {
+    if (supportIterable) {
+      return Iterable.class.isAssignableFrom(rawType);
+    }
+    return Collection.class.isAssignableFrom(rawType);
```

Comment:
Why not check for both? That would also eliminate the need to register `CollectionTypeAdapterFactory` twice.
```suggestion
    return Collection.class.isAssignableFrom(rawType) || (supportIterable && Iterable.class.isAssignableFrom(rawType));
```

Reply:
Well... I usually avoid complex condition checks like that: I find them harder to read, understand and maintain. Why I prefer the single `if` above is that I can read it literally _if the special mode is enabled, then just tell whether it's supported; otherwise use the default check_ whereas I read yours as _check if it's a collection, and if it's not then check if the special mode is enabled, and then check if it's an iterable_ (+ one more `isAssignable` call for the "special" mode). Additionally, it's more flexible not reducing into a single boolean result, and does not hit Git diff insertion/deletion rate.

Reply:
Nevermind, since `Collection extends Iterable` this check is fine.